checks/evaluation/branch_protection.go

```diff
@@ -373,18 +373,37 @@ func adminReviewProtection(branch *clients.BranchRef, dl checker.DetailLogger) (
 	// Only log information if the branch is protected.
 	log := branch.Protected != nil && *branch.Protected
 
-	if branch.BranchProtectionRule.CheckRules.UpToDateBeforeMerge != nil {
+	if branch.BranchProtectionRule.CheckRules.UpToDateBeforeMerge == nil ||
+		branch.BranchProtectionRule.RequireLastPushApproval == nil {
+		debug(dl, log, "unable to retrieve whether up-to-date branches are needed to merge on branch '%s'", *branch.Name)
+		return score, max
+	}
+
+	// Process UpToDateBeforeMerge value.
+	if branch.BranchProtectionRule.CheckRules.UpToDateBeforeMerge == nil {
+		debug(dl, log, "unable to retrieve whether up-to-date branches are needed to merge on branch '%s'", *branch.Name)
```

Comment:
Isn't this code unreachable based on L379's return? Same for L398, that's also unreachable because of the return. Should the `||` on L376 be `&&`?

Reply:
Yep, thanks.

Reply:
Thanks for the review!